### DISCUSSION TITLE
Migrations not being able to down

### DISCUSSION THREAD

**Discussion by Mvzundert** [spatie/laravel-ciphersweet#5] [Q&A] [answered]
Since the migrations only has an "up" function and no "down" function they break when you're doing local development with this package installed, I was wondering if this was by design or an oversight and if it is by design how to deal with the migrations not being executed every time you reseed the database.

I can imagine it slowing down migrations having to re-index every time, but now having this package installed means you can't do

`php artisan migrate:refresh --seed`

since it breaks (the table already exists).

**Comment by riasvdv** [comment, marked as answer]
We don't use `down` migrations, you can use `php artisan migrate:fresh --seed` instead

**Reply by Mvzundert** [reply]
Like described in my first post, when using that command, it doesn't "remove" the table so when the migration gets executed again it breaks because it's trying to create a table that already exists... If you don't want to use a down migration I feel like a check (if the table exists don't try to recreate it) on the up would be in order.

I realise some people don't use down on their migrations, but sometimes it's needed and having this package make the "refresh" option unavailable seems unwanted.

**Reply by riasvdv** [reply]
`migrate:fresh` will delete the tables, `migrate:refresh` doesn't indeed

**Reply by Mvzundert** [reply]
@riasvdv so that means by installing this package you lose the ability to refresh entirely, which like I said, seems weird when a simple statement in the up function (checking if the table exists and if so not doing anything) solves this issue.

**Reply by freekmurze** [reply]
We only use `migrate:fresh`. If you need support for `migrate:refresh`, fork the package and adapt it to your particular needs.

**Reply by Mvzundert** [reply]
@freekmurze So this package basically breaking a core Laravel functionality means if I want to use it I need to fork it and fix that myself? If this package was only for use within Spatie I'd get the policy but since this is a public package it shouldn't break core Laravel functionality (IMHO)

Off course you guys are welcome to use and create the package as you see fit.

**Reply by freekmurze** [reply]
From here on out it's better for you to assume that all packages are only intended to use within Spatie, because we indeed only build them for ourselves. If others can use them, great. If not, we don't consider that our problem. Everyone is free to fork our open source packages and adapt them to their particular liking.

**Reply by Mvzundert** [reply]
@freekmurze I get that, I only ran into this issue because of the fresh command dropping ALL tables (regardless of prefix) in a database that is shared by Laravel and another app which refresh doesn't do. If anything i hope that people that run into this issue at least know to fork this package and solve the issue.

Hence me starting this discussion.
